filtering.tex

```diff
@@ -1,90 +1,28 @@
 \chapter{Filtering} \label{ch:filtering}
 
-
-The instruction trace encoder must be able to filter on the following inputs to the encoder:
+Filtering provides a mechanism to control whether the encoder should produce trace.  For example,
+it may be desirable to trace:
 
 \begin{itemize}
-  \item The instruction address 
-  \item The context 
-  \item The exception cause 
-  \item Whether the exception is an interrupt or not 
-  \item The privilege level
-  \item Tval 
-  \item User specific signals
+  \item When the instruction address is within a particular range;
+  \item Starting from one instruction address and continuing until a second instruction address;
+  \item For one or more specified privilege levels;
+  \item For a particular context or range of contexts;
+  \item Exception and/or interrupt handlers for specified exception causes or with particular \textbf{tval} values;
+  \item Based on values applied to the \textbf{impdef} or \textbf{trigger} signals;
+  \item For a fixed period of time
+  \item etc.
 \end{itemize}
 
-Internal to the encoder will be several comparators and filters. The
-actual number of these will vary for different classes of devices. The
-filters and comparators must be configured to provide the trace and
-filtering required. There will be three command types needed to set up
-the filtering operation.
-
-\begin{enumerate}
-    \item Set up comparator 
-    \begin{itemize}
-      \item Which input bus to compare
-        \begin {enumerate}
-          \item address
-          \item context
-          \item tval
-        \end{enumerate}
-      \item Which comparator(s) to use which filtering operation to enable
-        \begin {enumerate}
-          \item \textit {eq}
-          \item \textit {neq}
-          \item \textit {lt}
-          \item \textit {lte}
-          \item \textit {gt}
-          \item \textit {gte}
-          \item \textit {always}
-          \end{enumerate}
-    \end{itemize}
-
-  \item Value e.g. start address
-    \item Set up filter 
-    \item Set match 
-    \begin{itemize}
-      \item Configure matching behaviour for exception, privilege and user sideband
-    \end{itemize}
-\end{enumerate}
+How this is accomplished is implementation specific.  
 
-The user may wish to:
+One suggested implementation provides: 

```

Comment:
This really should point to the debug module trigger support section in ingressPort.tex since this suggested implementation lacks specificity.

Reply:
Agreed - I'll fix.